### ISSUE TITLE
Debug LiteServ build fails: missing symbol _CBLRegisterJSViewCompiler

### ISSUE THREAD

**Issue by snej** [couchbase/couchbase-lite-ios#1063]
Debug builds of LiteServ fail with a link error for a missing symbol `_CBLRegisterJSViewCompiler`. This symbol is part of the CBL API but recent build changes have caused it to disappear from the debug build of the CBL framework.

**Comment by snej** [comment]
Earlier today I tried to fix this by making the debug Mac build of CBL.framework use the .exp file. This fixes the problem because the .exp file contains `_CBLRegisterJSViewCompiler` and forces that symbol to not get stripped. Unfortunately it causes all symbols _not_ in the .exp file to be hidden, and that utterly breaks the unit tests which need to call all sorts of internal functions/classes. So I had to back it out.

**Comment by snej** [comment]
The problem appears to be that
* CBLRegisterJSViewCompiler() is defined all by itself in CBLRegisterJSViewCompiler.m, with no other symbols in that file.
* CBL itself never calls that function; it's just there for an app to call if it needs to use JS views.
* We are now pre-building the CBL sources into a static library (.a file) in another target.

The result is that we run into a behavior in the linker where, if it sees that none of the symbols from a particular object file in a static library are being used, it doesn't load that object file at all. (This is sort of like the problem that the `-ObjC` linker flag fixes, except that flag only works on Obj-C categories, not functions.)

The workaround is to add the `-all_load` linker flag, which disables the behavior described above. I added this to the linker flags (for debug builds _only_), and it fixed the problem. Yay!

Unfortunately it added a link error building CBL.framework: duplicate definitions of `_CouchbaseLiteVersionNumber`. This happens because both the CBL Mac Library and the CBL Mac targets have versioning turned on, which creates a source file defining that symbol. I turned off versioning for the CBL Mac Library target, and it fixed the problem.
